week3/predict_citibike.Rmd

```diff
@@ -0,0 +1,167 @@
+}
+plot_data <- data.frame(K, train_err, validate_err) %>%
+  gather("split", "error", -K)
+ggplot(plot_data, aes(x=K, y=error, color=split)) +
+  geom_line() +
+  scale_x_continuous(breaks=K) +
+  xlab('Polynomial Degree') +
+  ylab('RMSE')
+```
+Models are similar and will move forward with attempting our linear models
+
+
+
+
+Will include other columns to the linear model 
+
+```{r}
+  
+  #since tmin and tmax are probably super correlated, i'll only include tmax
+  # fit on the training data
+model <- lm(num_trips ~ tmax + snwd + prcp + snow + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+
+train_err
+validate_err
+```
+Appears to have lower validation error than the previous graph's lowest point. 
+
+```{r}
+tidy(model)
+```
+Using this tidy model, we have determined most p values are appropriate but it appears that snow doesn't have a strong enough influence and will drop in next model. 
+
+```{r}
+model <- lm(num_trips ~ tmax + snwd + prcp + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+train_err
+validate_err
+```
+Error rose after excluding snow, but not enough so we will keep this model. 
+
+
+
+
+Note that temperature has the most effect and as a result will use the poly() at degree 2 to represent tmax. 
+```{r}
+model <- lm(num_trips ~ poly(tmax, 2, raw = T)
+ + snwd + snow + prcp + is_holiday + weekday, data=trips_per_day_train)
+    
+  # rmse on the training data
+train_err <- sqrt(mean((predict(model, trips_per_day_train) - trips_per_day_train$num_trips)^2))
+  # rmse on the validate data
+validate_err <- sqrt(mean((predict(model, trips_per_day_validate) - trips_per_day_validate$num_trips)^2))
+train_err
+validate_err
+```
+Error appears slightly better and will retain this model and use on test data. 
+
+
+
+
+```{r}
+trips_per_day_train <- trips_per_day_train %>%
+  add_predictions(model) %>%
+  mutate(split = "train")
+
+trips_per_day_validate <- trips_per_day_validate %>%
+  add_predictions(model) %>%
+  mutate(split = "validate")
+
+plot_data <- bind_rows(trips_per_day_train, trips_per_day_validate)
+ggplot(plot_data, aes(x = ymd, y = num_trips)) +
+  geom_point(aes(color = split)) +
+  geom_line(aes(y = pred)) +
+  xlab('Day') +
+  ylab('Number of Trips') 
+```
+
+Although the it appears to represent some overfit, that could be a result of weekday factors since travel is dependent upon day of the week. 

```

Comment:
Yeah it probably just looks like that because "time" is not really part of your model, so there's other relationships, implicit and explicit that are being exaggerated.